qutebrowser/browser/webengine/webenginesettings.py

```diff
@@ -434,29 +434,30 @@ def _init_site_specific_quirks():
                "Safari/{webkit_version} "
                "Edg/{upstream_browser_version}")
 
-    user_agents = {
+    user_agents = [
         # Needed to avoid a ""WhatsApp works with Google Chrome 36+" error
         # page which doesn't allow to use WhatsApp Web at all. Also see the
         # additional JS quirk: qutebrowser/javascript/quirks/whatsapp_web.user.js
         # https://github.com/qutebrowser/qutebrowser/issues/4445
-        'https://web.whatsapp.com/': no_qtwe_ua,
+        ("ua-whatsapp", 'https://web.whatsapp.com/', no_qtwe_ua),
 
         # Needed to avoid a "you're using a browser [...] that doesn't allow us
         # to keep your account secure" error.
         # https://github.com/qutebrowser/qutebrowser/issues/5182
-        'https://accounts.google.com/*': edge_ua,
+        ("ua-google", 'https://accounts.google.com/*', edge_ua),
 
         # Needed because Slack adds an error which prevents using it relatively
         # aggressively, despite things actually working fine.
         # September 2020: Qt 5.12 works, but Qt <= 5.11 shows the error.
         # https://github.com/qutebrowser/qutebrowser/issues/4669
-        'https://*.slack.com/*': new_chrome_ua,
-    }
+        ("ua-slack", 'https://*.slack.com/*', new_chrome_ua),
+    ]
 
-    for pattern, ua in user_agents.items():
-        config.instance.set_obj('content.headers.user_agent', ua,
-                                pattern=urlmatch.UrlPattern(pattern),
-                                hide_userconfig=True)
+    for name, pattern, ua in user_agents:
+        if name not in config.val.content.site_specific_quirks.skip:
+            config.instance.set_obj('content.headers.user_agent', ua,
+                                    pattern=urlmatch.UrlPattern(pattern),
+                                    hide_userconfig=True)
 
     config.instance.set_obj(
```

Comment:
This should also have a name FWIW

Reply:
What do you mean ?  putting this ``` config.instance.set_obj('content.headers.user_agent', ua,
                                    pattern=urlmatch.UrlPattern(pattern),
                                    hide_userconfig=True)``` in a variable ? adding name in set_obj ? (if so set_obj takes only these parameters)

Reply:
Setting the empty `accept_language` header for `matchmaker.krunker.io` (this line, not the user agent things above) is also a site-specific quirk (added recently in 7de06df5f4d034c50f96f0d5a241d5e222cb5d2a). Right now, it's impossible to turn that one off individually, because it's the only quirk which doesn't have a name.

So this should also be guarded with a `if 'misc-krunker' not in config.val.content.site_specific_quirks.skip:` or so, with that also being a valid value for the setting.